packages/widgets/src/blocks/widget-box/edit.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalUseInnerBlocksProps as useInnerBlocksProps,
+	useBlockProps,
+	InnerBlocks,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { HEADING_PLACEHOLDER } from './constants';
+
+const TEMPLATE = [ HEADING_PLACEHOLDER ];
+
+export default function Edit() {
+	const innerBlocksProps = useInnerBlocksProps(
+		{
+			className: 'wp-widget-box__inner-blocks',
+		},
+		{
+			template: TEMPLATE,
+			renderAppender: InnerBlocks.ButtonBlockAppender,
```

Comment:
If necessary we could create our own appender (probably in a follow up) which could be more specific about the fact that you should insert a Widget.